src/scripts/main.js

```diff
@@ -1,3 +1,265 @@
 'use strict';
 
-// write your code here
+const scoreDisplay = document.querySelector('.game-score');
+const button = document.querySelector('.start');
+const cells = document.querySelectorAll('td');
+const messageWin = document.querySelector('.message-win');
+const messageLose = document.querySelector('.message-lose');
+const messageStart = document.querySelector('.message-start');
+let score = 0;
+
+button.addEventListener('click', e => {
+  startGame();
+  score = 0;
+  scoreDisplay.innerHTML = 0;
+  document.addEventListener('keyup', move);
+});
+
+function startGame() {
+  cells.forEach(el => {
+    el.innerHTML = '';
+
+    messageWin.classList.add('hidden');
+    messageLose.classList.add('hidden');
+    messageStart.classList.add('hidden');
+  });
+  button.classList.remove('restart-lose');
+  button.className = 'button restart';
+  button.innerHTML = 'Restart';
+  generate();
+  generate();
+  styles();
+}
+styles();
+
+function generate() {
+  const randomNumber = Math.floor(Math.random() * cells.length);
+
+  if (cells[randomNumber].innerHTML === '') {
+    cells[randomNumber].innerHTML = random();
+    checkForLose();
+  } else {
+    generate();
+  }
+}
+
+function random() {
+  return Math.random() > 0.9 ? 4 : 2;
+}
+
+function moveRight() {
+  for (let i = 0; i < 16; i++) {
+    if (i % 4 === 0) {
+      const totalOne = cells[i].innerHTML;
+      const totalTwo = cells[i + 1].innerHTML;
+      const totalThree = cells[i + 2].innerHTML;
+      const totalFour = cells[i + 3].innerHTML;
+      const row = [+totalOne, +totalTwo, +totalThree, +totalFour];
+
+      const filtredRow = row.filter(num => num);
+      const missing = 4 - filtredRow.length;
+      const zeros = Array(missing).fill('');
+
+      const newRow = zeros.concat(filtredRow);
+
+      cells[i].innerHTML = newRow[0];
+      cells[i + 1].innerHTML = newRow[1];
+      cells[i + 2].innerHTML = newRow[2];
+      cells[i + 3].innerHTML = newRow[3];
+    }
+  }
+}
+
+function moveLeft() {
+  for (let i = 0; i < 16; i++) {
+    if (i % 4 === 0) {
+      const totalOne = cells[i].innerHTML;
+      const totalTwo = cells[i + 1].innerHTML;
+      const totalThree = cells[i + 2].innerHTML;
+      const totalFour = cells[i + 3].innerHTML;
+      const row = [+totalOne, +totalTwo, +totalThree, +totalFour];
+
+      const filtredRow = row.filter(num => num);
+      const missing = 4 - filtredRow.length;
+      const zeros = Array(missing).fill('');
```

Comment:
`moveRight ` function has same code - create separete function that will be do it and return result array